app/controllers/posts_controller.rb

```diff
@@ -237,6 +238,20 @@ def destroy_many
     render nothing: true
   end
 
+  def merge_posts
+    params.require(:post_ids)
+
+    posts = Post.where(id: post_ids_including_replies).order(:id)
+    raise Discourse::InvalidParameters.new(:post_ids) if posts.blank?
```

Comment:
If `post_ids_including_replies` is equal to `[1, 2, 3]` where only `[1, 2]` are valid ids. The check here won't catch that `3` is invalid because we are only checking if `posts` collection is blank.